### PR TITLE
Bugfix outputs influxdb endless retires

### PR DESCRIPTION
**Pull request by MyaLongmire**
### Required for all PRs:

- [ x ] Updated associated README.md. - no update needed
- [ x ] Wrote appropriate unit tests. - no new unit test needed

[outputs.influxdb writing to missing databases or retentions result in endless retries #9211](https://github.com/influxdata/telegraf/issues/9211)

Fixed the repeating 404: database not found error by checking for the error and that skipping database creation is set to true in the config file and then printing an error and returning.

Fixed the repeating 500: Interal Server Error: no retention policy error by handling it as an API error.

### CODE REVIEW

File: plugins/outputs/influxdb/http.go [hunk 1]
```diff
@@ -317,7 +318,6 @@ func (c *httpClient) Write(ctx context.Context, metrics []telegraf.Metric) error
 					c.config.URL, dbrp.Database, err)
 			}
 		}
-
```
Comment (helenosheaa) [inline plugins/outputs/influxdb/http.go:320]: add back in the new line here

Comment (helenosheaa) [inline plugins/outputs/influxdb/http.go:320]: add back in new line

File: plugins/outputs/influxdb/http.go [hunk 2]
```diff
@@ -374,6 +374,12 @@ func (c *httpClient) writeBatch(ctx context.Context, db, rp string, metrics []te
 		desc = writeResp.Err
 	}
 
+	//checks for any 4xx code and drops metric and retrying will not make the request work
+	if len(resp.Status) > 0 && resp.Status[0] == '4' && c.config.SkipDatabaseCreation {
+		c.log.Errorf("E! [outputs.influxdb] Failed to write metric (will be dropped: %s): %s\n", resp.Status, desc)
+		return nil
+	}
```
Comment (helenosheaa) [inline plugins/outputs/influxdb/http.go:381]: I'm wondering if we should handle this case within the database not found error handling which is influxdb.go file from line 222. As the error is the same "database not found", but we weren't handling the case where SkipDatabaseCreation is true.

**Review by helenosheaa — COMMENTED** [review]
Thanks for the PR! Just a question around the location of one part of the logic.

**Review by sspaink — APPROVED** [review]

**Pull request merged**